apps/shared/y4m.c

```diff
@@ -201,6 +202,76 @@ static int y4mReadLine(FILE * inputFile, avifRWData * raw, const char * displayF
     return -1;
 }
 
+// Limits each sample value to fit into avif->depth bits and avif->yuvRange.
```

Comment:
1. Optional: Since the AV1 encoder and decoder do not care whether the samples are full range or limited range, this function can ignore `avif->yuvRange` and clamp the samples based on `avif->depth` only. An advantage is that this will make the issue I noted at line 220 and line 251 moot.

2. Another design choice is for the function to fail (rather than clamp) if it detects an out-of-range sample.

Reply:
> 1. Optional: Since the AV1 encoder and decoder do not care whether the samples are full range or limited range, this function can ignore `avif->yuvRange` and clamp the samples based on `avif->depth` only.

Done.

> An advantage is that this will make the issue I noted at line 220 and line 251 moot.

`avifRange` should still be documented in avif.h.

> 2. Another design choice is for the function to fail (rather than clamp) if it detects an out-of-range sample.

I believe being permissive on input is convenient for most users, at the expense of behavior transparency, which is mitigated by the warning.